discover/eval.sh

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env bash
+
+set -e
+
+declare JSON
+
+function eval() {
+  echo "::group::Nix Evaluation"
+
+  local system
+
+  system="$(nix eval --raw --impure --expr 'builtins.currentSystem')"
+  JSON="$(nix eval "$FLAKE#__std.ci.$system" --json)"
+
```

Comment:
Here we need to filter out the ones that have already been built (and are in cache).

I guess we can apply the simplifying assumption that actions are (supposed to be) idempotent.

Reply:
I don't think this is the right place, since even if some of them are cached, certain jobs we still want to run regardless. The build step already figures out what's already cached and skips those things.

Reply:
Since we already are giving special semantics to `action == 'build'`, we can at least filter those out. That should get us 80% saving, already.